api/include/opentelemetry/trace/noop.h

```diff
@@ -9,22 +16,32 @@ namespace opentelemetry
 namespace trace
 {
 /**
- * Noop implementation of Span.
+ * No-op implementation of Span. This class should not be used directly.
  */
 class NoopSpan final : public Span
 {
 public:
   explicit NoopSpan(const std::shared_ptr<Tracer> &tracer) noexcept : tracer_{tracer} {}
 
-  // Span
+  void AddEvent(nostd::string_view name) override {}
+
+  void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) override {}
+  void AddEvent(nostd::string_view name, core::SteadyTimestamp timestamp) override {}
+
+  void UpdateName(nostd::string_view name) override {}
+
+  void End() override {}
+
+  bool IsRecording() const override { return false; }
+
   Tracer &tracer() const noexcept override { return *tracer_; }
 
 private:
   std::shared_ptr<Tracer> tracer_;
 };
 
 /**
- * Noop implementation of Tracer
+ * No-op implementation of Tracer. This class should not be used directly.
```

Comment:
Why shouldn't we use a no-op tracer directly?

Reply:
I assumed a binary using otel wouldn't explicitly create or set or use the NoopTracer. What's the use-case for it? (I removed the comment either way)

Reply:
I could imagine users wanting to use a different ownership model than the TracerProvider that creates one tracer per named component and ties lifetime to a global.

Envoy, for example, uses one tracer per thread
https://github.com/envoyproxy/envoy/blob/master/source/extensions/tracers/lightstep/lightstep_tracer_impl.cc#L156

Reply:
That makes sense.

What I was trying to do is steer users away from the NoopTracer specifically, and encouraging them to use the TracerProvider to get a tracer.